portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,9 +32,8 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     ArrayList<String> messages = new ArrayList<String>();
```

Comment:
Initialize `messages` outside of both doGet() and doPost() functions. Only do this one.